### ISSUE TITLE
Button: The Saving animation doesn't respect prefers-reduced-motion

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#55545]
### Description

Various buttons in the editor display an animation when saving.

This animation doesn't respect the OS-level "Reduce Motion" user preference.

I'm marking this issue as high priority as the speific animation in use is ery close to a 'flashing colors' effect that could trigger seizure. This animation is pontentially dangerous for WordPress users and should be fixed soon.

I'm not going to attach animated GIFs to illustrate the animation, as they could be potentially dangerous for users reading this GitHub issue. I'll attach a fre static images:

![gray low contrast](https://github.com/WordPress/gutenberg/assets/1682452/9c0b802d-460b-4526-90c0-67de3e47c7f2)

![blue](https://github.com/WordPress/gutenberg/assets/1682452/6e5e6631-091e-4e5e-b758-dcb7a1ba0d92)

![Screenshot 2023-10-23 at 14 24 52](https://github.com/WordPress/gutenberg/assets/1682452/afa215de-5fd9-4739-aba6-b326e22c7275)

Additionally, the gray 'saving' button clearly has an insufficient color contrast ratio.

### Step-by-step reproduction instructions

- Enable the 'Reduce motion' preference in your operaring system.
- Edit a post.
- Click the Update button.
- See the button grradient animation runs, despite the 'Reduce motion' preference.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
For reference, the gray button insufficient color contrast ratio is as follows:
```
text color: #949494
first background gradient color: #fafafa
Contrast Ratio: 2.91:1

text color: #949494
second background gradient color: #e0e0e0
Contrast Ratio: 2.3:1
```

The text color is inherited from `.edit-site-save-hub`.
